util/util.go

```diff
@@ -86,6 +86,33 @@ func FileMetaEqual(actual data.FileMeta, expected data.FileMeta) error {
 	return nil
 }
 
+func BytesMatchLenAndHashes(fetched []byte, length int64, hashes data.Hashes) error {
+	flen := int64(len(fetched))
+	if length != 0 && flen != length {
+		return ErrWrongLength{length, flen}
+	}
+
+	if len(hashes) != 0 {
```

Comment:
Should this error if there are no hashes?

Reply:
I don't think so, length and hashes are both optional for [`METAFILES`](https://theupdateframework.github.io/specification/latest/#metafiles).